modules/miscutil/lib/upgrades/invenio_2015_01_13_hide_holdings.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2014 CERN.
+#
+# Invenio is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the Free Software Foundation,
+# Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""Upgrade for explicitly hiding the 'holdings' tab for all collections\
+ except 'Books' without an existing rule"""
+
+from invenio.dbquery import run_sql
+
+depends_on = ['invenio_release_1_1_0']
+
+
+def info():
+    """Upgrade recipe information."""
+    return "Updates the collectiondetailedrecordpagetabs to hide 'holdings' for every collection \
+            except 'Books' without an existing rule."
+
+
+def do_upgrade():
```

Comment:
In Invenio, when applying settings to subcollections, an entry is made into the database for each subcollection as well as the main collection. I.E. every collection (demo or custom) that has any settings other than default (show all) will not be touched by this upgrade script, due to the `GNORE` statement in the SQL query which will avoid inserting rules for collection IDs already found in the `collectiondetailedrecordpagetabs` table.